src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs

```diff
@@ -54,6 +55,30 @@ public sealed class DotNetCorePublishSettings : DotNetCoreSettings
         /// </remarks>
         public bool NoRestore { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to force all dependencies to be resolved even if the last restore was successful. This is equivalent to deleting project.assets.json.
+        /// </summary>
+        /// <remarks>
+        /// Requires .NET Core 2.x or newer.
+        /// </remarks>
+        public bool Force { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether Publish the .NET Core runtime with your application so the runtime doesn't need to be installed on the target machine. Defaults to 'true' if a runtime identifier is specified.
+        /// </summary>
+        /// <remarks>
+        /// Requires .NET Core 2.x or newer.
+        /// </remarks>
+        public bool SelfContained { get; set; }
+
+        /// <summary>
+        /// Gets or sets the specified NuGet package sources to use during the restore.
+        /// </summary>
+        /// <remarks>
+        /// Requires .NET Core 2.x or newer.
+        /// </remarks>
+        public ICollection<string> Sources { get; set; }
```

Comment:
I think it would be a good idea to initialize this in the constructor to avoid null reference exceptions if the user assumes it's not null.

Reply:
That would make it inconsistent with i.e.
https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs#L19

https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/NuGet/Restore/NugetRestoreSettings.cs#L32

though.

Reply:
But not inconsistent with https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs#L149